internal/controllers/snapshot_controller.go

```diff
@@ -169,8 +169,22 @@ func (r *SnapshotReconciler) removeSnapshot(log logr.Logger, snapshotID string,
 	}
 	log.V(2).Info("Snapshot references", "pools", len(pools), "rbd-images", len(imgs))
 
-	if len(pools) != 0 || len(imgs) != 0 {
-		return fmt.Errorf("unable to delete snapshot: still in use")
+	for _, clonedImgName := range imgs {
+		log.V(2).Info("Open cloned image to flatten", "ClonedImageID", clonedImgName)
+		img, err := librbd.OpenImage(ioCtx, clonedImgName, librbd.NoSnapshot)
+		if err != nil {
+			if !errors.Is(err, librbd.ErrNotFound) {
+				return fmt.Errorf("failed to open cloned image: %w", err)
+			}
+			continue
+		}
+		defer closeImage(log, img)
+
+		log.V(2).Info("Flatten cloned image", "ClonedImageID", clonedImgName)
+		if err := img.Flatten(); err != nil {
```

Comment:
As flattening an image adds considerable time consumption we may want to limit parallel flattening operations.